app/src/main/java/fr/free/nrw/commons/media/MediaClient.java

```diff
@@ -123,7 +140,12 @@ public MediaClient(MediaInterface mediaInterface, MediaDetailInterface mediaDeta
                     || null == mwQueryResponse.query().pages()) {
                 return Observable.empty();
             }
-            continuationStore.put(key, mwQueryResponse.continuation());
+            if(mwQueryResponse.continuation() != null) {
+                continuationStore.put(key, mwQueryResponse.continuation());
```

Comment:
if null couldn't we just remove the key and then it all works as usual without the extra map

Reply:
If we remove the key then it would be like fetching the first page. The API call would be made without the continuation params.

Reply:
And then the Db update would return the same data, not retriggering a boundaryCallback as the list would be unchanged, I think.

Though I guess now the api is set up with a continuation... I am quite unsure of how to proceed

Reply:
and the continuation map is never assigned but I am sure you are on that

Reply:
> Though I guess now the api is set up with a continuation... I am quite unsure of how to proceed

Yeah, right. As the API is set up with a continuation it will start fetching all the pages again. On a related note, the current logic of fetching pages(that I added in this PR) is very naive and can be improved. For now, i wanted to keep things simple. 

> and the continuation map is never assigned but I am sure you are on that

I didn't understand this comment.

Reply:
I should have said `continuationExists` map but it is resolved now.

This is a bit "upending the teatable" but it seems to me that this endpoint shouldn't use continuation at all, we could happily query using

![image](https://user-images.githubusercontent.com/3358282/82648690-f2ca2d00-9c0f-11ea-8829-e9d92e9d69e0.png)

but perhaps that is best suited for another ticket

Reply:
Yeah we can use `aistart` and `aiend` parameter to fetch contributions based on timestamp but I don't think that can replace continuation. With start and end timestamp, there's no way to be sure how many contributions would be returned. For eg. recently I uploaded around 2000 images in a day. The API would then return all images in one call. It would defeat the purpose of pagination. 

To limit it to 10 images per call we will have to use `ailimit` and then `continuation` will still come into picture.

Reply:
@macgills Is everything working in this PR now? It would be great if you could do another test.

Reply:
It is a little trickier granted but if at top of list we request media since now and reorder it so it is oldest first it hoovers along nicely.

Bottom of the list request since last item timestamp + newest first.

Again perhaps a discussion for another ticket. I will get to testing shortly